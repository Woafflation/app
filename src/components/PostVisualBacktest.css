.post-visual-backtest {
  line-height: 16px;
  display: flex;
  flex-direction: column;
  padding: 20px;
  width: 260px;
  background: linear-gradient(
    180deg,
    var(--grey-light) 0%,
    rgba(255, 255, 255, 0) 100%
  );
  height: 100%;
  justify-content: space-between;
}

.post-visual-backtest__info {
  margin-bottom: 5px;
  display: flex;
}

.post-visual-backtest__changes {
  font-size: 14px;
  text-align: center;
  color: var(--grey-dark-2);
}

.post-visual-backtest__percent {
  font-size: 14px;
  text-align: right;
}

@media (min-width: 768px) {
  .post-visual-backtest {
    display: flex;
  }

  .post-visual-backtest__info {
    flex-direction: column;
  }
}
