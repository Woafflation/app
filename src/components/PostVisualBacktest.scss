@import "~@santiment-network/ui/mixins";

.post-visual-backtest {
  width: 231px;
  background: linear-gradient(180deg, var(--athens), hsla(0, 0%, 100%, 0));
  display: flex;
  flex-direction: column;
  text-align: right;

  @include responsive('phone', 'phone-xs') {
    width: 100%;
  }
}

.post-visual-backtest__info {
  margin: 9px auto auto auto;
  display: flex;
}

.post-visual-backtest__changes {
  font-size: 14px;
  text-align: center;
  align-items: center;
  color: var(--waterloo);
}

.post-visual-backtest__percent {
  font-size: 14px;
  margin-left: auto;
  position: relative;
  top: -10px;
}

@media (min-width: 768px) {
  .post-visual-backtest {
    display: flex;
  }

  .post-visual-backtest__info {
    flex-direction: column;
  }
}
