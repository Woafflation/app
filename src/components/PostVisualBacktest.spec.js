/* eslint-env jest */
import React from 'react'
import { shallow, mount, render } from 'enzyme'
import toJson from 'enzyme-to-json'
import { PostVisualBacktest } from './PostVisualBacktest'

describe('PostVisualBacktest component', () => {
  it('(smoke) it should render correctly', () => {
    const component = render(
      <PostVisualBacktest
        history={{ loading: false }}
        ticker='SAN'
        change={3}
        from='aksdfjh'
      />
    )
    expect(toJson(component)).toMatchSnapshot()
    expect(component.html()).toEqual(null)
  })

  it('SAN icon should render correctly', () => {
    const component = render(
      <PostVisualBacktest
        ticker='SAN'
        history={{ historyPrice: [], loading: false }}
        change={25}
        changeProp='priceUsd'
        from='aksdfjh'
      />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})
