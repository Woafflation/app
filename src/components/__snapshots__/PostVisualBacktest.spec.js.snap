// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostVisualBacktest component (smoke) it should render correctly 1`] = `null`;

exports[`PostVisualBacktest component SAN icon should render correctly 1`] = `null`;
